v2/stack_analyses.yaml

```diff
@@ -631,33 +594,14 @@ components:
       - $ref: '#/components/schemas/StackAnalysesResult'      
```

Comment:
why do we have a ref to stackanalysesresult when that already contains this object which is added via discriminator?

Reply:
Discriminator property just gives a hint of property/field which can be used to detect the type of a model. It doesn't actually includes the base model anyways,

Please check https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/

Reply:
It would really make sense to include either registered user or free tier in the stack analysis result using the discriminator and oneof.
The way it is currently done is that stack analysis result is composed inside stack analysis result for free tier. 
The result from the api should be one object which is composed differently based on the use case.

Reply:
There are 2 ways to use discriminator. 

1. Describe it in the BaseModel itself
2. Describe it in the place where Derived models are used.

I've used approach #1. For more details, Please check https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#discriminator-object